pkg/sdn/api/validation/validation.go

```diff
@@ -117,3 +117,29 @@ func ValidateNetNamespace(netnamespace *sdnapi.NetNamespace) field.ErrorList {
 func ValidateNetNamespaceUpdate(obj *sdnapi.NetNamespace, old *sdnapi.NetNamespace) field.ErrorList {
 	return validation.ValidateObjectMetaUpdate(&obj.ObjectMeta, &old.ObjectMeta, field.NewPath("metadata"))
 }
+
+// ValidateEgressNetworkPolicy tests if required fields in the EgressNetworkPolicy are set.
+func ValidateEgressNetworkPolicy(policy *sdnapi.EgressNetworkPolicy) field.ErrorList {
+	allErrs := validation.ValidateObjectMeta(&policy.ObjectMeta, true, oapi.MinimalNameRequirements, field.NewPath("metadata"))
+
+	for i, rule := range policy.Spec.Egress {
+		if rule.Type != sdnapi.EgressNetworkPolicyRuleAllow && rule.Type != sdnapi.EgressNetworkPolicyRuleDeny {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("egress").Index(i).Child("type"), rule.Type, "invalid policy type"))
+		}
+
+		_, _, err := net.ParseCIDR(rule.To.CIDRSelector)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("egress").Index(i).Child("to"), rule.To.CIDRSelector, err.Error()))
+		}
+	}
```

Comment:
"conflicting" how exactly? The rules are ordered, and it's expected that they may refer to overlapping ranges. (eg, "allow 192.168.1.0/24, deny 192.168.0.0/16") I guess we could check for impossible-to-match rules (like if those two were in the opposite order), but I don't think that kind of error is especially more likely than any other rule "typos" which we wouldn't be able to check ("oops, I meant 192.168.*2*.0") so I don't think it's really worth it.